File: aliapp/libray/Start.js
import {Post,Storage} from './ui/index'
import {MapGeolocation} from './plus/index'
import Socket from './Socket'

/* 启动 */
export default {

  self: null,

  /* 初始化 */
  init(self){
    this.self = self;
    /* 登录验证 */
    this.tokenState(1);
    clearInterval(this.tokenInterval);
    this.tokenInterval = setInterval(()=>{
      this.tokenState(0);
    },10000);
    /* 获取定位 */
    this.geoLocation();
    /* 消息推送 */
    Socket.start(this.self);
  },

  /* 登录验证 */
  tokenState(uinfo){
    const token = Storage.getItem('token');
    if(token){
      Post('user/token',{token:token,uinfo:uinfo},(res)=>{
        const d = res.data;
        if(d.code==0){
          this.self.store.data.isLogin = true;
          if(d.uinfo) this.self.store.data.uInfo = d.uinfo;
          this.self.update();
        }else{
          this.self.store.data.isLogin = false;
          this.self.store.data.uInfo = {};
          this.self.update();
          Storage.setItem('token','');
        }
      });
    }else{
      this.self.store.data.isLogin = false;
      this.self.update();
      Storage.setItem('token','');
    }
  },

  /* 获取定位 */
  geoLocation(){
    MapGeolocation((res)=>{
      this.self.store.data.geolocation = res;
      this.self.update();
      Storage.setItem('city',res.district);
    });
  },

}

File: flutter/lib/library/Start.dart
import 'package:flutter/material.dart';
import 'package:webmis/env.dart';
import 'package:webmis/library/ui/ui.dart';
import 'package:webmis/library/ui/storage.dart';
import 'package:webmis/library/ui/request.dart';
import 'package:webmis/library/inc/time-set.dart';
import 'package:webmis/library/plus/map-geolocation.dart';
import 'package:webmis/library/Socket.dart';

/* 启动 */
class Start{

  static BuildContext _context;
  static var _tokenInterval;

  /* 初始化 */
  static init(BuildContext context){
    _context = context;
    // 登录验证
    if(Env.login['start']){
      tokenState(1);
      if(_tokenInterval!=null) _tokenInterval.cancel();
      _tokenInterval = setInterval((res)=>tokenState(1),10000);
    }
    /* 获取定位 */
    if(Env.amap['start']) geoLocation();
    /* 消息推送 */
    if(Env.socket['start']) Socket.start(_context);
  }

  /* 登录验证 */
  static Future<void> tokenState(int uinfo) async {
    final token = await Storage.getItem('token');
    if(token!=null && token!=''){
      Map res = await post(Env.login['api'], {'token': token, 'uinfo':uinfo.toString()});
      if(res['code']==0){
        Ui.store(_context).setIsLogin(true);
        // 用户信息
        if(res[Env.login['uinfo']].runtimeType == Map){
          Ui.store(_context).setUInfo(res[Env.login['uinfo']]);
        }
      }else{
        Ui.store(_context).setIsLogin(false);
        Ui.store(_context).setUInfo({});
        await Storage.setItem('token','');
      }
    }else{
      Ui.store(_context).setIsLogin(false);
      await Storage.setItem('token',null);
    }
  }

  /* 获取定位 */
  static Future<void> geoLocation() async {
    MapGeolocation.fetch(_context).then((res){
      Ui.store(_context).setGeolocation(res);
      Storage.setItem('city',res['district']);
    }).catchError((e)=>print('定位失败！'));
  }

}